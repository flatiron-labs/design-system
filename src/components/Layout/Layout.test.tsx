import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import { Layout } from '.'

describe('Layout', () => {
  const text = 'Content'

  const renderComponent = () =>
    render(
      <Layout>
        <p>Content</p>
      </Layout>
    )

  it('should append the global styles', async () => {
    renderComponent()

    await waitFor(() => {
      expect(document.querySelector('style[data-styled="active"]')).toMatchSnapshot()
    })
  })

  it('should render children', () => {
    renderComponent()
    expect(screen.getByText(text)).toBeInTheDocument()
  })
})
