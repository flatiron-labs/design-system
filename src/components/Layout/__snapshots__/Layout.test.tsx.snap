// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should append the global styles 1`] = `
<style
  data-styled="active"
  data-styled-version="5.2.1"
>
  body{background-color:rgba(3,4,22,1);font-family:Fira Code;color:rgba(255,255,255,1);}
</style>
`;
